### PR TITLE
Fix Audio Mixer crash

### PR DESCRIPTION
**Pull request by Atlante45**
- Fixes a rare crash in the Audio Mixer (possibly happen in the Avatar Mixer too, but I have not observed it)

### CODE REVIEW

File: libraries/networking/src/LimitedNodeList.h [hunk 1]
```diff
@@ -197,8 +197,12 @@ class LimitedNodeList : public QObject, public Dependency {
                 *lockWaitOut = (endLock - start);
             }
 
-            std::vector<SharedNodePointer> nodes(_nodeHash.size());
-            std::transform(_nodeHash.cbegin(), _nodeHash.cend(), nodes.begin(), [](const NodeHash::value_type& it) {
```
Comment (Atlante45) [inline libraries/networking/src/LimitedNodeList.h]: The crash is caused if a node gets added after the call to `_nodeHash.size()` and before `std::transform` gets to the element directly after the one that was just inserted.

If that happens, `nodes` will be `n` elements long and `std::tranform` will iterate on `n + 1` element.
This results in the internal code of `std::transform` trying to dereference `nodes.end()` to assign a value to it, which is not possible.


I fixed the issue, by only reserving the guessed amount of space needed and back inserting the elements into `nodes`

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1435/HighFidelity-Beta-PR10910-0c18f04e58736cb39c6810be7b500b8d19afaa04.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1435/HighFidelity-Beta-PR10910-0c18f04e58736cb39c6810be7b500b8d19afaa04.exe

**Comment by Atlante45** [comment]
I stumbled into the crash while trying to repro the Avatar Mixer crash we've been seeing recently.

Repro:
- Run Sandbox
- Go to `localhost:40100` in your browser
- Click on the UUID for the Audio Mixer
- Open a new tab and do the same for the Avatar Mixer
- Open several Interfaces and run this script:
```js
var timer = Script.setInterval(function() {
    if (Window.location.hostname === "localhost") {
        Window.location = "hifi://poker";
    } else {
        Window.location = "hifi://localhost";
    }
}, 3000);

Script.scriptEnding.connect(function() {
    Script.clearInterval(timer);
});
```
- If you see the text in one of your tab turn red, this means the Mixer most likely crashed and that you've reproed the crash.
- I've personally tested with 4 Interfaces and it takes between 5 and 10 min for the crash to occur.

**Comment by CainFoool** [comment]
Testing.

**Comment by CainFoool** [comment]
# Test Passed
Test | Result
-------- | --------
Run Sandbox | Pass 
Go to localhost:40100 in your browser | Pass
Click on the UUID for the Audio Mixer | Pass
Open a new tab and do the same for the Avatar Mixer | Pass
Open several Interfaces and run this script: `...` | Pass

I was not able to reproduce the text text under the Mixer tabs **in this PR** - assuming this is a pass.

Let me know if this needs further tested.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

File: libraries/networking/src/LimitedNodeList.h (hunk 1 above)
Comment (jherico) [inline libraries/networking/src/LimitedNodeList.h]: I don't understand... if `_nodeHash` can be modified by another thread, neither the old code nor the new code are safe

Comment (Atlante45) [inline libraries/networking/src/LimitedNodeList.h]: This is a TBB concurrent map. We use a read lock for all the concurrently safe operations (Insert, iterate, size) and a write lock for all unsafe operation (erase).

The concurrency issue here is not the multiple accesses to the container, but rather the assumption that it's size will not change.

**Comment by Atlante45** [comment]
@CainFoool You shouldn't be able to repro on this PR, it fixes it. Can you try to repro on master?

**Comment by CainFoool** [comment]
@Atlante45: Reproduced on Master.
firefox_2017-07-07_20-02-52

Waiting around 20 minutes (just to make sure) the crash was happening on Master and not PR.

**Comment by Atlante45** [comment]
@CainFoool That's perfect, thanks!

**Pull request merged**